src/Stratis.Bitcoin.Tests/Consensus/ChainedHeaderTreeTest.cs

```diff
@@ -579,5 +581,60 @@ public void ChainHasOneCheckPointAndAssumeValid_ChainsWithCheckpointButMissedAss
             result.DownloadFrom.HashBlock.Should().Be(listOfCurrentChainHeaders.Skip(2).First().GetHash());
             result.DownloadTo.HashBlock.Should().Be(listOfCurrentChainHeaders.Last().GetHash());
         }
+
+        /// <summary>
+        /// Issue 17 @ We advanced consensus tip (CT) and there are some partially validated (PV) headers after the CT.
+        /// Now we receive headers that are after the last PV header and include assume valid (AV). Make sure that those
+        /// headers that are before the AV header and after the last PV are all marked as AV. 
+        /// </summary>
+        [Fact]
+        public void ChainHasPartiallyValidatedAfterConsensusTip_NewHeadersWithAssumeValidPresented_CorrectHeadersAreMarkedAsAssumedValid()
+        {
+            // Chain header tree setup.
+            // Initial chain has 4 header with the consensus tip at header 4.
+            // Example: fv1=fv2=fv3=fv4 (fv - fully validated).
+            const int initialChainSize = 4;
+            TestContext ctx = new TestContextBuilder().WithInitialChain(initialChainSize).Build();
+            ChainedHeaderTree cht = ctx.ChainedHeaderTree;
+            ChainedHeader initialChainTip = ctx.InitialChainTip;
+
+            // Extend the chain with 2 partially validated headers.
+            // Example: fv1=fv2=fv3=(fv4)=pv5=pv6 (pv - partially validated).
+            const int partiallyValidatedHeadersCount = 2;
+            initialChainTip = ctx.ExtendAChain(partiallyValidatedHeadersCount, initialChainTip);
```

Comment:
After this, the name of the variable is misleading. If you use just `chainTip` you will not have any confusion.